### PR TITLE
[WIP] Feat: add Vue integration

### PR DESCRIPTION
**Pull request by pedronauck**
### Description

This pull request support Vue integration

### Review

- [x] Add example with vue
- [x] Add docz-plugin-vue
- [ ] Translate VueWrapper to component name in output

### Pre-merge checklist

- [ ] Test `<PropsTable>` integration
- [x] Test `<Playground>` integration

### CODE REVIEW

File: packages/docz-plugin-vue/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { createPlugin } from 'docz-core'
+import { VueLoaderPlugin } from 'vue-loader'
+
+interface BabelRC {
```
Comment (pedronauck) [inline packages/docz-plugin-vue/src/index.ts]: @marceloavf pode remover essa interface daqui, não precisa dela porque o TS já inferi ela...

File: packages/docz-plugin-vue/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  presets?: any[]
+  plugins?: any[]
+  cacheDirectory?: boolean
+  babelrc?: boolean
+}
+
+export const doczPluginVue = () =>
+  createPlugin({
+    modifyBabelRc: (babelrc: BabelRC): BabelRC => {
```
Comment (pedronauck) [inline packages/docz-plugin-vue/src/index.ts]: ai você não precisa usar ela aqui também!

File: packages/docz-plugin-vue/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    modifyBundlerConfig: config => {
+      config.module.rules.push({
+        test: /\.vue$/,
+        use: 'vue-loader',
+        exclude: /node_modules/,
+      })
+
+      config.module.rules.push({
+        test: /\.css$/,
```
Comment (pedronauck) [inline packages/docz-plugin-vue/src/index.ts]: uma parada que tava pensando, é como integrar isso com o [docz-plugin-css](https://github.com/pedronauck/docz/tree/master/packages/docz-plugin-css) 🤔

File: examples/vue/src/components/Alert.vue [hunk 4]
```diff
@@ -0,0 +1,24 @@
+<template>
+  <div class="alert">
+    <slot></slot>
+    {{ kind }}
+  </div>
+</template>
+
+<script>
+  export default {
+    name: 'Alert',
```
Comment (pedronauck) [inline examples/vue/src/components/Alert.vue]: Tu consegue deixar esse Alert no mesmo padrão de alert dos outros exemplos? Mudando o background baseado nos kinds e tals?

File: packages/docz-plugin-vue/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+import { VueLoaderPlugin } from 'vue-loader'
+
+interface BabelRC {
+  presets?: any[]
+  plugins?: any[]
+  cacheDirectory?: boolean
+  babelrc?: boolean
+}
+
+export const doczPluginVue = () =>
```
Comment (pedronauck) [inline packages/docz-plugin-vue/src/index.ts]: aqui vamos manter o padrão de simplicidade, ao invés de deixar `doczPluginVue` pode deixar só `vue` ✌️

File: packages/docz-plugin-vue/src/index.ts (hunk 3 above)
Comment (marceloavf) [reply packages/docz-plugin-vue/src/index.ts]: That should be a good thing too, we could specify it in Playground tag or also in configuration file, in vue we can just set a tag `lang="scss"` to change the processor of `template`, `style` and `script`. Might be a way to get it. https://vue-loader.vuejs.org/guide/pre-processors.html#sass

**Comment by pedronauck** [comment]
Closed until we figure out the best strategy for that 👍

**Comment by marceloavf** [comment]
Still no solution for the parse of proptables, component name conversion and full compatibility tested with render and single file components @artalar.

**Comment by artalar** [comment]
What about https://github.com/akxcv/vuera ?

**Comment by marceloavf** [comment]
We are using `vuera` @artalar, you can see more in this branch https://github.com/pedronauck/docz/tree/feat/vue

We also have some problems with babel, as you can see here too https://github.com/akxcv/vuera/issues/46

If you figure out something, feel free to discuss or help us with that integration!

**Comment by Aaron-Pool** [comment]
So, would these problems also exist for plain old `.js` Vue components? My company uses Vue, but our design system code is purely `.js`, we don't use `.vue` single file components. We just write Vue components with a render function and JSX. Its very react-y, actually.

**Comment by marceloavf** [comment]
We need also to make pure render components to work @Aaron-Pool, they both render normally, but we still have an issue with `PropsTable` and make `vuera` more interactive to render our code.

**Comment by Aaron-Pool** [comment]
Ok! I'll look through the vue branch you have set up and see if there's anything I can think of worth suggesting. Thanks for the quick response!

**Comment by Aaron-Pool** [comment]
Ok, so, just a thought after poking around for a bit, instead of using `VueWrapper`, have you considered creating a simple webpack loader to chain after the `.vue` file loader that  does two things:

1) Runs the component through `vue-docgen-api` (https://github.com/vue-styleguidist/vue-docgen-api)
2) Wraps the vue component in the vuera `VueInReact` HOC
3) Assign the output of `vue-docgen-api` to __docgenInfo, just like `react-docgen` does

Then you could conditionally skip adding `react-docgen` loader to jsx if any non-compatible plugins have been required (such as a vue plugin), so that the `__docgenInfo` isn't overwritten?

Or, if you didn't want to worry about conditionally skipping the `react-docgen` loader based on current plugins, you could just assign the output of `vue-docgen-api` to `__altDocgenInfo` and use `__altDocgenInfo` if it's defined when you parse the props/component name. That could be a pattern for future "nonReact" jsx plugins as well.

**Comment by marceloavf** [comment]
Great ideas @Aaron-Pool, I was thinking about a while ago about abandon `VueWrapper` to make it more interactive and show real component name inside playground, but I can't make it work with webpack.

Also saw `vue-docgen-api` and tried something, but no success either. Some of this points can also be discussed with @pedronauck, since him understand more of the `docz-core`

I'll try to rebase the `feat/vue` branch to get it working with latest release of docz, I would be very grateful if you could help me implement these solutions to see what we can get.

Thank you again!

**Pull request closed without merging**